### PR TITLE
MueLu: regionMG replacing Map with const Map where possible to avoid casting

### PR DESCRIPTION
**Pull request by pohm01**
@trilinos/muelu

## Motivation
Making changes suggested in #8437 

## Testing
Region tests run locally and produce same result as before these changes.

### CODE REVIEW

File: packages/muelu/research/regionMG/src/SetupRegionHierarchy_def.hpp
Comment (mayrmt) [inline packages/muelu/research/regionMG/src/SetupRegionHierarchy_def.hpp:305]:
```suggestion
    RCP<const Map> regRowMapCurrent = regProlong->getColMap();
```

File: packages/muelu/research/regionMG/src/SetupRegionHierarchy_def.hpp
Comment (mayrmt) [inline packages/muelu/research/regionMG/src/SetupRegionHierarchy_def.hpp:358]: @pohm01 Feel free to change the interface of `regionalToComposite()`, if this helps to avoid casts.

**Review by mayrmt — APPROVED** [review]
@pohm01 Looks good. I removed the `AT: auto merge` flag and leave it up to you to merge right away or to do some more changes.

File: packages/muelu/research/regionMG/src/SetupRegionHierarchy_def.hpp
Comment (pohm01) [reply packages/muelu/research/regionMG/src/SetupRegionHierarchy_def.hpp:358]: This involves reworking the region unit tests to use const maps which I think I will do as a separate PR

**Review by mayrmt — APPROVED** [review]
@pohm01 Thanks for your contribution!

**Review by lucbv — APPROVED** [review]
Yes, look's good to me : )

**Pull request merged**
